domain/externalcontroller/modelmigration/export_test.go

```diff
@@ -0,0 +1,202 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package modelmigration
+
+import (
+	"context"
+
+	"github.com/golang/mock/gomock"
+	"github.com/juju/description/v4"
+	"github.com/juju/errors"
+	"github.com/juju/names/v4"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/core/crossmodel"
+)
+
+type exportSuite struct {
+	coordinator *MockCoordinator
+	service     *MockExportService
+}
+
+var _ = gc.Suite(&exportSuite{})
+
+func (s *exportSuite) setupMocks(c *gc.C) *gomock.Controller {
+	ctrl := gomock.NewController(c)
+
+	s.coordinator = NewMockCoordinator(ctrl)
+	s.service = NewMockExportService(ctrl)
+
+	return ctrl
+}
+
+func (s *exportSuite) newExportOperation() *exportOperation {
+	return &exportOperation{
+		service: s.service,
+	}
+}
+
+func (s *exportSuite) TestExportExternalController(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	modelUUID := "model-uuid-1"
+	dst := description.NewModel(description.ModelArgs{})
+	dst.AddRemoteApplication(description.RemoteApplicationArgs{
+		SourceModel: names.NewModelTag(modelUUID),
+	})
+	ctrlUUID := "ctrl-uuid-1"
+	extCtrlModel := []crossmodel.ControllerInfo{
+		{
+			ControllerTag: names.NewControllerTag(ctrlUUID),
+			Addrs:         []string{"192.168.1.1:8080"},
+			Alias:         "external ctrl1",
+			CACert:        "ca-cert-1",
+			ModelUUIDs:    []string{"model1", "model2"},
+		},
+	}
+	s.service.EXPECT().ControllersForModels(gomock.Any(), []string{modelUUID}).
+		Times(1).
+		Return(extCtrlModel, nil)
+
+	// Assert that the destination description model has no external
+	// controllers before the migration:
+	c.Assert(dst.ExternalControllers(), gc.HasLen, 0)
+	op := s.newExportOperation()
+	err := op.Execute(context.Background(), dst)
+	c.Assert(err, jc.ErrorIsNil)
+	// Assert that the destination description model has one external
+	// controller after the migration:
+	c.Assert(dst.ExternalControllers(), gc.HasLen, 1)
```

Comment:
Generally, `Assert` is used if it needs to be so to avoid subsequent verifications from panicking. Prefer `Check` when otherwise, because although the test will fail, you get more complete validation from a single run.

Reply:
done, see 95564b14e4c4efb7653904e05c7eeab5ab1bbe9b